python/discord/bot.py

```diff
@@ -0,0 +1,149 @@
+import discord
+from discord.ext import commands
+import datetime
+import os
+import sys
+import dotenv
+import asyncio
+import time
+from dotenv import load_dotenv
+dir_path = os.path.dirname(os.path.realpath(__file__))
+parent_dir_path = os.path.abspath(os.path.join(dir_path, os.pardir))
+sys.path.insert(0, parent_dir_path)
+from connection import Topic, connecter
+
+
+load_dotenv()
+
+TOKEN = os.getenv('DISCORD_TOKEN')
+
+mqtt = connecter.create("Discord Bot")
+
+t = Topic()
+
+VELOCITY = 50
+DIRECTION = 50
+
+client = commands.Bot(command_prefix=".", help_command=None)
+
+@client.command()
+async def help(ctx):
+    embed = discord.Embed(colour=discord.Colour(
+        0xf5a623), timestamp=datetime.datetime.utcnow())
+    embed.add_field(name=".drive", value="the drive command takes a string of < (left), > (right), f (forward), r (reverse), b (boost), s (stop) and does each of them for one second. \n[Example: .drive <f>b>s>f]", inline=False)
+    embed.add_field(name=".go", value="forward", inline=False)
+    embed.add_field(name=".gor", value="forward & right", inline=False)
+    embed.add_field(name=".gol", value="forward & left", inline=False)
+    embed.add_field(name=".rr", value="reverse", inline=False)
+    embed.add_field(name=".rrl", value="reverse & left", inline=False)
+    embed.add_field(name=".rrr", value="reverse & right", inline=False)
+    embed.add_field(name=".s", value="stop", inline=False)
+    await ctx.send(embed=embed)
+
+# def seperateArgs(args, delimeter):
+#     finalArgs = []
+#     toAppend = ''
+#     index = 0
+#     for i in args:
+#         if(i == delimeter):
+#             finalArgs.append(toAppend.strip())
+#             toAppend = ''
+#         else:
+#             toAppend += i
+#         if(index == len(args) - 1):
+#             finalArgs.append(toAppend.strip())
+#             toAppend = ''
+#         index += 1
+#     return finalArgs
+
+
+async def control(v, d):
+  if v > 0:
+    mqtt.publish(t.throttle_forward, str(v))
+  elif v < 0:
+    mqtt.publish(t.throttle_reverse, str(v))
+  else:
+    mqtt.publish(t.throttle_reverse, "0")
+  if d > 0:
+    mqtt.publish(t.steering_right, str(d))
+    time.sleep(1)
+    mqtt.publish(t.steering_right, "0")
+  elif d < 0:
+    mqtt.publish(t.steering_left, str(d))
+    time.sleep(1)
```

Comment:
Suggestion to have the drive command have 0.3 sleep timer instead, as it takes mulitple commands at once, so the user can see the input faster.